### ISSUE TITLE
Sqlite errors

### ISSUE THREAD

**Issue by SteVwonder** [flux-framework/flux-core#730]
I was running a "large" simulation with ~2k jobs.  After a bit, I got the following error immediately following a job completing (which triggered code within my simulator to run and modify the kvs):
```
2016-07-13T00:49:44.544611Z content-sqlite.err[0]: store: executing stmt: disk I/O error(778)
2016-07-13T00:49:44.544699Z broker.err[0]: content store: Input/output error
2016-07-13T00:49:44.544714Z broker.debug[0]: content flush begin
2016-07-13T00:49:44.545946Z broker.debug[0]: content flush +1 (dirty=3 pending=3)
2016-07-13T00:49:44.546381Z content-sqlite.err[0]: store: executing stmt: disk I/O error(778)
2016-07-13T00:49:44.546462Z broker.err[0]: content store: Input/output error
2016-07-13T00:49:44.546472Z broker.debug[0]: content flush begin
2016-07-13T00:49:44.547059Z broker.debug[0]: content flush +1 (dirty=2 pending=2)
2016-07-13T00:49:44.546547Z content-sqlite.err[0]: store: executing stmt: disk I/O error(778)
2016-07-13T00:49:44.547169Z broker.err[0]: content store: Input/output error
2016-07-13T00:49:44.547179Z broker.debug[0]: content flush begin
2016-07-13T00:49:44.547767Z broker.debug[0]: content flush +1 (dirty=2 pending=2)
2016-07-13T00:49:44.547934Z content-sqlite.err[0]: store: executing stmt: database disk image is malformed(11)
2016-07-13T00:49:44.547991Z broker.err[0]: content store: Invalid argument
```

The simulation continues to make progress, but the last 4 lines continually repeat (with the dirty and pending numbers increasing).

I am running with the persist-filesystem set to a directory on lustre: "/p/lcratery/herbein1/persist_dir".

I am re-running the simulation with the persist-filesystem set to the tmp nfs filesystem to see if that makes a difference.

**Comment by SteVwonder** [comment]
The sqlite db leftover in the persist content directory is exactly 512M (536870912 bytes).  This seems like it is not a coincidence and is instead some limitation within sqlite that I'm bumping into.

**Comment by SteVwonder** [comment]
Confirmed that NFS filesystem does not help.  It again broke when the db reach 536870912 bytes.

**Comment by garlick** [comment]
That's odd, do you have ulimit -f set to some value other than unlimited?  I know I've pushed many gigabytes into that thing in the past....

**Comment by SteVwonder** [comment]
@garlick: good though, but unfortunately not.  It is unlimited:
```
[21:47:52] herbein1 at hype356 in /nfs/tmp2/herbein1/flux-persist/fluxP-92869-87vcvC/content
$ ulimit -S -f
unlimited

[21:47:53] herbein1 at hype356 in /nfs/tmp2/herbein1/flux-persist/fluxP-92869-87vcvC/content
$ ulimit -H -f
unlimited
```

**Comment by garlick** [comment]
I'll try tomorrow but can you create a file larger than that with `dd if=/dev/zero bs=1024k count=513 of=test_file_name`?

**Comment by SteVwonder** [comment]
@garlick: Yep

```
[21:47:55] herbein1 at hype356 in /nfs/tmp2/herbein1/flux-persist/fluxP-92869-87vcvC/content
$ dd if=/dev/zero bs=1024k count=513 of=test_file_name
513+0 records in
513+0 records out
537919488 bytes (538 MB) copied, 4.85198 s, 111 MB/s

[21:54:51] herbein1 at hype356 in /nfs/tmp2/herbein1/flux-persist/fluxP-92869-87vcvC/content
$ ls -lah
total 1.1G
drwxr-xr-x 2 herbein1 herbein1 2.0K Jul 12 21:54 .
drwx------ 3 herbein1 herbein1 2.0K Jul 12 21:20 ..
-rw-r--r-- 1 herbein1 herbein1 512M Jul 12 21:29 sqlite
-rw-r--r-- 1 herbein1 herbein1 513M Jul 12 21:54 test_file_name
```

**Comment by grondo** [comment]
Some other thoughts before nighty night time.
* Do you get past this limit using local fs like /tmp for sqlite file?
* We were testing simple wreckrun of thousands of jobs before. Do you still
hit IO errors at 512m in this simple scenario?
* I think searching online real quick I saw someone had this problem due to
a fd leak. Though I don't understand it, does increasing all fd limit help
at all? (Grasping now)

Sorry if I am no help.

**Comment by SteVwonder** [comment]
I apologize profusely.  @garlick you hit the nail on the head with your first message.  I was checking the ulimit on hype356 but running the tests on a tmux window with an allocation on hype350.  Setting ulimit to unlimited fixes the problem.  Thanks guys!  Sorry for the noise and wild goose chase.

**Comment by garlick** [comment]
No worries, but just in case this is an issue lurking for others to hit, can you track down where your ulimit is being set?   I removed my `.nodotkit` file on hype and re-logged in, and my ulimit -f is unlimited both at login and if I srun a shell script that runs ulimit -f.  I don't seem to be setting this  to unlimited in any of my dot files.

**Comment by SteVwonder** [comment]
It turns out that I shot myself in the foot.  I ran `ulimit -cS $(( 1024 * 1024 ))` to increase the size of my core dump, but the ordering of arguments should be `-Sc`.  The `-cS` ordering caused ulimit to discard the arguments and to assume I meant `-f`.  Hence my ulimit issue only occurring in 1 tmux window.  I have learned some valuable lessons: `-f` is a thing (I was unaware of this), drop the `-S`, and `unlimited` is much shorter and less error prone than `$(( 1024 * 1024 ))`.

Thanks again for the late night help.
